File: src/main/java/net/softbell/bsh/dto/request/MemberDto.java
package net.softbell.bsh.dto.request;

import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import net.softbell.bsh.domain.BanRule;
import net.softbell.bsh.domain.MemberRole;
import net.softbell.bsh.domain.entity.Member;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 회원정보 DTO
 */
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MemberDto
{
    private Long memberId;
    private Integer ban;
	private Date banDate;
	private Date changePasswdDate;
	private String email;
	private Date lastLogin;
	private Integer loginFailCount;
	private String nickname;
	private String password;
	private Integer permission;
	private Date registerDate;
	private String userId;
	private String username;

    public Member toEntity()
    {
    	// Generate
        return Member.builder()
                .userId(userId)
        		.email(email)
                .password(password)
                .name(username)
        		.nickname(nickname)
                .registerDate(new Date())
                .ban(BanRule.ofLegacyCode(ban))
                .loginFailCount(loginFailCount)
                .permission(MemberRole.ofLegacyCode(permission))
                .build();
    }
}

File: src/main/kotlin/net/softbell/bsh/dto/request/MemberDto.kt
package net.softbell.bsh.dto.request

import net.softbell.bsh.domain.BanRule
import net.softbell.bsh.domain.MemberRole
import net.softbell.bsh.domain.entity.Member
import java.util.*

/**
 * @author : Bell(bell@softbell.net)
 * @description : 회원정보 DTO
 */
class MemberDto(
        //    var memberId: Long = null
        var ban: Int = 0,
        var banDate: Date?,
        var changePasswdDate: Date?,
        var email: String,
        var lastLogin: Date?,
        var nickname: String,
        var password: String,
        var permission: Int = 0,
        var registerDate: Date = Date(),
        var userId: String,
        var username: String
) {
    fun toEntity(): Member {
        // Generate
        return Member(
                userId = userId,
                email = email,
                password = password,
                name = username,
                nickname = nickname,
                registerDate = registerDate,
                ban = BanRule.ofLegacyCode(ban),
                permission = MemberRole.ofLegacyCode(permission)
        )
    }
}